### ISSUE TITLE
Could not load type 'System.Windows.Threading.DispatcherObject' from assembly 'WindowsBase'

### ISSUE THREAD

**Issue by chanallenk** [windows-toolkit/Microsoft.Toolkit.Win32#220]
## Describe the bug
There is an issue with publishing Window Store apps with Xaml Islands.

After installing a published Windows Store application with Xaml Islands using the Windows Application Packaging Project, the app will not launch. 

I was able to reproduce this with both a simple application using .NET Core 3.0 as shown in the documentation, and a large project using .NET Core 3.1, both with Xaml Islands and custom controls.

Event Log:

Application: WPFCoreCustomXAMLIsland.exe
CoreCLR Version: 4.700.19.51502
.NET Core Version: 3.0.1
Description: The process was terminated due to an unhandled exception.
Exception Info: System.TypeLoadException: Could not load type 'System.Windows.Threading.DispatcherObject' from assembly 'WindowsBase, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

Faulting application name: WPFCoreCustomXAMLIsland.exe, version: 1.0.0.0, time stamp: 0x5da7a666
Faulting module name: KERNELBASE.dll, version: 10.0.18362.535, time stamp: 0x50cc8d5a
Exception code: 0xe0434352
Fault offset: 0x000000000003a839
Faulting process id: 0x10adc
Faulting application start time: 0x01d5b308811b8ac2
Faulting application path: C:\Program Files\WindowsApps\56113TestSoftware.XamlTest_1.0.6.0_x64__agy8jafheqyng\WPFCoreCustomXAMLIsland.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 05f11ca4-6ffc-48a2-ae5a-6763c40d76e9
Faulting package full name: 56113TestSoftware.XamlTest_1.0.6.0_x64__agy8jafheqyng
Faulting package-relative application ID: App

## Steps to Reproduce
1. Create or use an existing simple Xaml Island application. I followed the exact same instructions [here](https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/host-custom-control-with-xaml-islands)
2. Publish the application as a Microsoft Store app (appx)
3. Install and attempt to launch the published application using the powershell script
4. App will crash immediately and logs to event viewer

## Expected behavior
No errors occur and app launches without issue

## Environment

```
NuGet Package(s): 
1. Microsoft.Toolki.Wpf.UI.XamlHost
2. Microsoft.VCRTForwarders.140

Package Version(s): 
1. v6.0.0
2. v1.0.4

Project .NET Version:
- [ ] .NET Framework (version: )
- [ ] .NET Core 3
- [X] .NET Core 3.1

Windows 10 Build Number:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [ ] May 2019 Update (18362)
- [X] November 2019 Update (18363)
- [ ] Insider Build (build number: )

App min and target version:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [X] May 2019 Update (18362)
- [ ] Insider Build (xxxxx)

Device form factor:
- [X] Desktop
- [ ] Xbox
- [ ] Surface Hub
- [ ] IoT

Visual Studio 
- [ ] 2017 (version: )
- [X] 2019 (version: 16.4.1) 
- [ ] 2019 Preview (version: )

```

**Comment by r2d2rigo** [comment]
Chiming in to say that I'm experiencing the same issue.

**Comment by r2d2rigo** [comment]
@chanallenk update - seems to be an issue with Visual Studio https://github.com/microsoft/Xaml-Islands-Samples/issues/5#issuecomment-577586584

**Comment by chanallenk** [comment]
@r2d2rigo Thank you for the update!

**Comment by michalleptuch** [comment]
Any update? Workaround? I reproduce this bug every time when I try to publish XAML Island app. I also try to use .NET Core 3.1 instead of .NET Core 3.0 and then I get another exception (check this issue: [https://github.com/microsoft/Xaml-Islands-Samples/issues/5#issuecomment-647009203](https://github.com/microsoft/Xaml-Islands-Samples/issues/5#issuecomment-647009203)).

How to reproduce both exceptions? Just get official repo ([https://github.com/microsoft/Xaml-Islands-Samples](https://github.com/microsoft/Xaml-Islands-Samples)) and publish WPF packaged app.

@chanallenk opened this issue last year and Microsoft does nothing! 😒

**Comment by chanallenk** [comment]
@michalleptuch From what I heard in the [other post](https://github.com/microsoft/Xaml-Islands-Samples/issues/5#issuecomment-577586584), this is strictly a tooling issue with Visual Studio. It seems the VS team is going to pretend our issues don't exist until .NET 5/WinUI 3.0 comes out. 

For now, I recommend rolling back to Visual Studio 2019 16.3.8 and using .NET Core 3.0. I've managed to successfully publish Xaml Island apps to the store with that build. Every release after that version seems to have broken everything with store publishing/association.

**Comment by chanallenk** [comment]
This appears to be resolved with updated toolkit v6.1.1.
